packages/fx-core/src/plugins/resource/sql/plugin.ts

```diff
@@ -271,6 +281,9 @@ export class SqlPluginImpl {
   }
 
   public async generateArmTemplates(ctx: PluginContext): Promise<Result<any, FxError>> {
+    if (ctx.answers?.existingResources?.includes(Constants.pluginFullName)) {
```

Comment:
minor, can we see which is better?
1. Check existingResources in plugin.ts as what we're doing here
2. Check existingResources in `generateArmTemplates` in index.ts, and call addNewDatabase() from there directly.

Reply:
Sure. 
Move the check in index.ts.